### DISCUSSION TITLE
Tabby doesn't load correctly/at all in nvim

### DISCUSSION THREAD

**Discussion by carrierg** [nanozuki/tabby.nvim#131] [Q&A]
Hi. I'm trying to figure out why tabby isn't loading properly in nvim. I'm visually impaired so I freely admit the error is very likely self-inflicted but my diagnostics aren't returning anything and I've read through this so many times by now.

Every one of my plugins loads and functions correctly except for tabby. Every plugin has its own lua file in my plugins folder, which is also where tabby.lua lives. Also have devicons installed, naturally. Color theme is dracula and I have nerd fonts enabled, yet nothing happens when I call for the tabline, as seen in the screenshot attached.

Would love to get this working because I want to switch from bufferline to tabby if I can. Treating individual buffers as tabs is madness IMHO.

Code is copy-pasted from the setup instructions:

```
return
{
  'nanozuki/tabby.nvim',
  dependencies = 'nvim-tree/nvim-web-devicons',
  config = function()
local theme = {
  fill = 'TabLineFill',
  -- Also you can do this: fill = { fg='#f2e9de', bg='#907aa9', style='italic' }
  head = 'TabLine',
  current_tab = 'TabLineSel',
  tab = 'TabLine',
  win = 'TabLine',
  tail = 'TabLine',
}
require('tabby.tabline').set(function(line)
  return {
    {
      { '  ', hl = theme.head },
      line.sep('', theme.head, theme.fill),
    },
    line.tabs().foreach(function(tab)
      local hl = tab.is_current() and theme.current_tab or theme.tab
      return {
        line.sep('', hl, theme.fill),
        tab.is_current() and '' or '',
        tab.number(),
        tab.name(),
        tab.close_btn(''),
        line.sep('', hl, theme.fill),
        hl = hl,
        margin = ' ',
      }
    end),
    line.spacer(),
    line.wins_in_tab(line.api.get_current_tab()).foreach(function(win)
      return {
        line.sep('', theme.win, theme.fill),
        win.is_current() and '' or '',
        win.buf_name(),
        line.sep('', theme.win, theme.fill),
        hl = theme.win,
        margin = ' ',
      }
    end),
    {
      line.sep('', theme.tail, theme.fill),
      { '  ', hl = theme.tail },
    },
    hl = theme.fill,
  }
end)
end,
}
```
![tabby](https://github.com/nanozuki/tabby.nvim/assets/107744715/c3844e95-5c97-44b9-af43-a63ce29961f2)

**Comment by nanozuki** [comment]
Your config works on my computer; wait for me for a day, and I will create a playground(reproduce) script to help debug the config.

**Reply by carrierg** [reply]
Thanks so much. I really can't figure out what's going on. I'm running lualine as my status bar but as far as I'm aware this shouldn't pose a problem based on what I read about tabby. If you need further details, let me know.

EDIT: If it's relevant, my init.lua is based on [kickstart.nvim](https://github.com/nvim-lua/kickstart.nvim/blob/master/init.lua). I've enabled line 285 to allow me to use the plugins folder as detailed in my original note above. Absolutely no problems with any other plugins so not sure why tabby's not cooperating....

**Reply by carrierg** [reply]
Figured it out! Apparently tabby doesn't like the plugin [taboo.vim](https://github.com/gcmt/taboo.vim), which I had been using prior to switching to Lua.

I'll have to find another plugin to force renaming of my tabs, unless tabby has that capability built into it as well?

Thank you once again for your time and patience.

**Reply by nanozuki** [reply]
`tabby.nvim` has a feature to rename the [names of tabs](https://github.com/nanozuki/tabby.nvim?tab=readme-ov-file#tab_name). And you should use it in your tabline setting, I found that you already add `tab.name()` in your config, so, that ok.

The problem is that there is no content "name of tab" in vim/nvim, plugins define the name of tabs. If you use multiple table plugins, that will cause conflict. You also use 'taboo.vim' with 'tabby.nvim'?

**Comment by nanozuki** [comment]
@notkerus I added a playground in tabby.nvim, you can check this to try your config: https://github.com/nanozuki/tabby.nvim?tab=readme-ov-file#playground

**Reply by carrierg** [reply]
@nanozuki, Yup, the problem was me. I had used taboo.vim when I was using Vim as I wasn't using Vim's native tabs there. I didn't see your instructions about how to rename a tab in tabby so I left 'taboo' in my plugin folder when I moved to nVim and Lua. Removing 'taboo' and using your renaming instructions in tabby works absolutely perfectly.

So if this helps anyone else: do NOT run taboo.vim with tabby as tabby has tab renaming functionality built in.

Wonderful plugin - thanks for making it!
